src/backend/distributed/utils/metadata_cache.c

```diff
@@ -3143,18 +3143,19 @@ GetLocalGroupId(void)
 										  tupleDescriptor, &isNull);
 
 		groupId = DatumGetInt32(groupIdDatum);
+
+		/* set the local cache variable */
+		LocalGroupId = groupId;
 	}
 	else
 	{
-		elog(ERROR, "could not find any entries in pg_dist_local_group");
+		/* upgrade is happening */
```

Comment:
When does that happen? With a concurrent metadata sync?

Reply:
When upgrading postgres, it seems this table is empty before `citus_finish_pg_upgrade()` finishes exeuction (https://github.com/citusdata/citus/blob/4f60e3a149fb83204278f50df9315828cffa8a28/src/backend/distributed/sql/udfs/citus_finish_pg_upgrade/9.0-1.sql#L1)

Reply:
also updated the comment.